modules/apps/captcha/captcha-impl/src/test/java/com/liferay/captcha/internal/function/captcha/FunctionCaptchaImplTest.java

```diff
@@ -0,0 +1,202 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.captcha.internal.function.captcha;
+
+import com.liferay.captcha.internal.configuration.FunctionCaptchaImplConfiguration;
+import com.liferay.portal.catapult.PortalCatapult;
+import com.liferay.portal.json.JSONArrayImpl;
+import com.liferay.portal.json.JSONObjectImpl;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+
+import java.util.concurrent.Future;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+/**
+ * @author Manuele Castro
+ */
+public class FunctionCaptchaImplTest {
+
+	@ClassRule
+	@Rule
```

Comment:
Hey @manuelecastro, could you apply the same pattern we have applied here (com.liferay.saml.web.internal.struts.SamlLoginActionTest)?

 Basically, we extracted the mocking code to external methods and called them in the setUp method.

Thx